File: src/main/java/demo/Null.java
String name = getCustomer().getName();  // Can be null
if (name != null) {
  System.out.println("Customer name: " + name);
} else {
  System.out.println("Customer name unavailable");
}

//
// Java 8
//

Optional<String> name = Optional.ofNullable(getCustomer().getName());
if (name.isPresent()) {
  System.out.println("Customer name: " + name.get());
} else {
  System.out.println("Customer name unavailable");
}

File: src/main/kotlin/demo/Null.kt
val name = getCustomer()?.name  // Returns null if getCustomer() is null
if (name != null) {
  println("Customer name: $name")
} else {
  println("Customer name unavailable")
}


// 
// Elvis Operator
//

val name = getCustomer()?.name ?: "Customer name unavailable"
println(name)